kinder/pkg/cluster/manager/actions/kubeadm-upgrade.go

```diff
@@ -131,6 +131,9 @@ func kubeadmUpgradeApply(c *status.Cluster, cp1 *status.Node, upgradeVersion *K8
 			applyArgs = append(applyArgs, "--patches", constants.PatchesDir)
 		}
 	}
+	if len(featureGates) > 0 {
+		applyArgs = append(applyArgs, fmt.Sprintf("--feature-gates=%s", featureGates))
```

Comment:
Success
```
 ./kubeadm upgrade apply -f v1.28.0-alpha.1.25+627f43d8bb6ce3 --v=6 --feature-gates="UpgradeAddonsBeforeControlPlane=true"
```

Failed with the same error
```
./kubeadm upgrade apply -f v1.28.0-alpha.1.25+627f43d8bb6ce3 --v=6 --feature-gates=\"UpgradeAddonsBeforeControlPlane=true\"
```

Reply:
Why does escaping happen here?

Reply:
I know the resaon, we need to do this:
```
        // We remove the leading and trailing double or single quotes because the
	// feature-gate could be set as
	// --kubeadm-feature-gate="RootlessControlPlane=true" or
	// --kubeadm-feature-gate='RootlessControlPlane=true', so the value
	// of featureGate string would be "\"RootlessControlPlane=true"\" or "RootlessControlPlane=true'" respectively.
	// Once we trim the value double or single quotes the value will be "RootlessControlPlane=true".
	trimmedFeatureGate := strings.Trim(featureGate, "\"'")
```